pandas/core/groupby/generic.py

```diff
@@ -1710,3 +1715,97 @@ def _normalize_keyword_aggregation(kwargs):
         order.append((column,
                       com.get_callable_name(aggfunc) or aggfunc))
     return aggspec, columns, order
+
+
+def _make_lambda(
```

Comment:
Do we really need this? Seems like this could be done a lot more succinctly with a partial, i.e.:

```python
f = functools.partial(func)
f.__name__ = "<lambda_{}>".format(i)
```

Could be done directly in loop so one less function and has the added benefit of maintaining other function attributes

Reply:
Do you find it strange to use `functors.partial` when you aren't partially applying any arguments? It's not where my mind initially went.

Reply:
Yea I can see where that is strange (effectively using `partial` to copy the function) but personally find it to be less code and more readable than this wrapper.

Reply:
I find the currently construction more informative (e.g. it has a doc-string and typing)